test/test_images/probe_helper/probe_helper.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2020 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type cloudEventsFunc func(cloudevents.Event) protocol.Result
+
+type envConfig struct {
+	// Environment variable containing the sink URL (broker URL) that the event will be forwarded to by the probeHelper
+	BrokerURL string `envconfig:"K_SINK" default:"http://default-brokercell-ingress.cloud-run-events.svc.cluster.local/cloud-run-events-probe/default"`
+
+	// Environment variable containing the port which listens to the probe to deliver the event
+	ProbePort int `envconfig:"PROBE_PORT" default:"8070"`
+
+	// Environment variable containing the port to receive the event from the trigger
+	ReceiverPort int `envconfig:"RECEIVER_PORT" default:"8080"`
+
+	// Environment variable containing the timeout period to wait for an event to be delivered back (in seconds)
+	Timeout int `envconfig:"TIMEOUT" default:"300"`
+}
+
+func forwardFromProbe(c cloudevents.Client, receivedEvents map[string] chan bool, timeout int) cloudEventsFunc{
+	return func(event cloudevents.Event) protocol.Result{
+		eventID := event.ID()
+		log.Print("Sending the cloud event to the broker")
+		receivedEvents[eventID] = make(chan bool, 1)
+		ctx, _ := context.WithTimeout(context.Background(), time.Duration(timeout) * time.Second)
+		if res := c.Send(ctx, event); !cloudevents.IsACK(res) {
+			log.Fatalf("Failed to send cloudevent: %v", res)
```

Comment:
Fatal will cause the process to exist. Let's return the response directly if there is an error, or return NACK. What status code is NACK?

Reply:
It returns http 500.